src/backend/base/langflow/components/vectorstores/chroma.py

```
@@ -85,8 +86,33 @@ class ChromaVectorStoreComponent(LCVectorStoreComponent):
+        ),
     ]
 
+    @override
+    def set_attributes(self, params: dict):
+        super().set_attributes(params)
+        raw_filter = params.get("search_filter", "")
+        if raw_filter:
+            try:
+                self.advance_search_filter = json.loads(raw_filter)
+            except json.JSONDecodeError:
+                raise ValueError("The 'advance_search_filter' must be a valid JSON dictionary.")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)